app/lib/methods/loadMessagesForRoom.ts

```diff
@@ -36,25 +35,27 @@ export default function loadMessagesForRoom(args: {
 	t: RoomTypes;
 	latest: string;
 	loaderItem: TMessageModel;
-}): Promise<IMessage[] | []> {
+}): Promise<(IMessageFromServer | ILoadMoreMessage)[]> {
 	return new Promise(async (resolve, reject) => {
 		try {
-			const data = await load(args);
+			const data: IMessageFromServer[] = await load(args);
 			if (data?.length) {
+				const loadMoreMessages: ILoadMoreMessage[] = [];
 				const lastMessage = data[data.length - 1];
 				const lastMessageRecord = await getMessageById(lastMessage._id);
 				if (!lastMessageRecord && data.length === COUNT) {
-					const loadMoreItem = {
+					const loadMoreMessage: ILoadMoreMessage = {
 						_id: generateLoadMoreId(lastMessage._id),
 						rid: lastMessage.rid,
-						ts: moment(lastMessage.ts).subtract(1, 'millisecond'),
+						ts: moment(lastMessage.ts).subtract(1, 'millisecond').toString(),
 						t: MESSAGE_TYPE_LOAD_MORE,
 						msg: lastMessage.msg
 					};
-					data.push(loadMoreItem);
+					loadMoreMessages.push(loadMoreMessage);
 				}
-				await updateMessages({ rid: args.rid, update: data, loaderItem: args.loaderItem });
-				return resolve(data);
+				// @ts-ignore - How does the update work if the decrypt gives deconstruct on prop that doesn't exist?
+				await updateMessages({ rid: args.rid, update: [...data, ...loadMoreMessages], loaderItem: args.loaderItem });
+				return resolve([...data, ...loadMoreMessages]);
```

Comment:
How about this?
```ts
const data: (IMessageFromServer | ILoadMoreMessage)[] = await load(args);
if (data?.length) {
	// const loadMoreMessages: ILoadMoreMessage[] = [];
	const lastMessage = data[data.length - 1];
	const lastMessageRecord = await getMessageById(lastMessage._id);
	if (!lastMessageRecord && data.length === COUNT) {
		const loadMoreMessage: ILoadMoreMessage = {
			_id: generateLoadMoreId(lastMessage._id),
			rid: lastMessage.rid,
			ts: moment(lastMessage.ts).subtract(1, 'millisecond').toString(),
			t: MESSAGE_TYPE_LOAD_MORE,
			msg: lastMessage.msg
		};
		data.push(loadMoreMessage);
	}
	// @ts-ignore - How does the update work if the decrypt gives deconstruct on prop that doesn't exist?
	await updateMessages({ rid: args.rid, update: data, loaderItem: args.loaderItem });
	return resolve(data);
```

We're going to be building the array with both types anyway.

Reply:
Good idea